darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,524 @@
+    print("filesystem_roots:", filesystem_roots)
+    for row, filesystem in enumerate(filesystem_roots):
+        ax_filesystem_bytes = fig.add_subplot(len(filesystem_roots), 
+                                              2,
+                                              row * 2 + 1)
+        ax_filesystem_counts = fig.add_subplot(len(filesystem_roots), 
+                                              2,
+                                              row * 2 + 2)
+        # hide the right side plot frame (spine) so that
+        # the value labels don't overlap with the plot frame
+        for axis in [ax_filesystem_bytes, ax_filesystem_counts]:
+            axis.spines['right'].set_visible(False)
+
+        print("filesystem:", filesystem)
+        print("bytes_rd_series:", bytes_rd_series)
+        print("bytes_wr_series:", bytes_wr_series)
+        # convert to MiB using the factor suggested
+        # by Google (approximate result only for now)
+        bytes_read = bytes_rd_series[filesystem]/1.049e+6
+        bytes_written = bytes_wr_series[filesystem]/1.049e+6
+
+        # scale to fit longer filesystem
+        # strings on the left side of the plots
+        # NOTE: may need more sophisticated scaling
+        # eventually
+        if len(filesystem) <= 8:
+            fontsize = 18
+        else:
+            fontsize = 12
+
+        ax_filesystem_bytes.annotate(filesystem, (-0.3, 0.5), fontsize=fontsize, xycoords='axes fraction')
+        ax_filesystem_counts.barh(0, file_wr_series[filesystem], color='red', alpha=0.3)
+        ax_filesystem_counts.barh(1, file_rd_series[filesystem], color='blue', alpha=0.3)
+
+        ax_filesystem_bytes.text(0, 0.75, f' # bytes read ({bytes_read:.2E} MiB)', transform=ax_filesystem_bytes.transAxes)
+        ax_filesystem_bytes.text(0, 0.25, f' # bytes written ({bytes_written:.2E} MiB)', transform=ax_filesystem_bytes.transAxes)
+
+        if file_rd_series[filesystem] == 0:
+            ax_filesystem_counts.text(0, 0.75, ' 0 files read', transform=ax_filesystem_counts.transAxes)
+        else:
+            ax_filesystem_counts.text(0, 1, '# files read')
+            ax_filesystem_counts.text(file_rd_series[filesystem], 1, '  ' + str(file_rd_series[filesystem]))
+
+        if file_wr_series[filesystem] == 0:
+            ax_filesystem_counts.text(0, 0.25, ' 0 files written', transform=ax_filesystem_counts.transAxes)
+        else:
+            ax_filesystem_counts.text(0, 0, '# files written')
+            ax_filesystem_counts.text(file_wr_series[filesystem], 0, str(file_wr_series[filesystem]))
+
+        if bytes_written != 0:
+            ax_filesystem_bytes.barh(0, bytes_written, color='red', alpha=0.3)
+        else:
+            ax_filesystem_bytes.barh(0, bytes_written, color='red', alpha=0.0)
+
+        if bytes_read != 0:
+            ax_filesystem_bytes.barh(1, bytes_read, color='blue', alpha=0.3)
+        else:
+            ax_filesystem_bytes.barh(1, bytes_read, color='blue', alpha=0.0)
```

Comment:
This may not do the same thing, since the object is not placed on the canvas at all in the `0` value cases if I do that. I'll assume you've thought of that and checked the consequences (lack of test catching doesn't necessarily make it safe at the moment).

I've pushed this change in, so I'll mark it resolved.